src/Microsoft.SqlTools.ServiceLayer/Migration/MigrationService.cs

```diff
@@ -402,6 +405,50 @@ public void InitializeService(ServiceHost serviceHost)
             }
         }
 
+        internal async Task HandleSaveSkuRecommendationsRequest(
+            SaveSkuRecommendationsParams parameters,
+            RequestContext<SaveSkuRecommendationsResult> requestContext)
+        {
+            const string SkuRecommendationJsonFileExt = ".json";
+            const string SkuRecommendationHtmlFileExt = ".html";
+            string timeStamp = DateTime.UtcNow.ToString("yyyyMMddHHmmss");
+
+            string[] skuRecommendationFileNames = new string[6];
+
+            try
+            {
+                // Save results
+                SkuRecommendationReport reportMI = new SkuRecommendationReport(
+                    new Dictionary<SqlInstanceRequirements, List<SkuRecommendationResult>> { [parameters.InstanceRequirements] = parameters.SqlMiRecommendationResults },
+                    AzureSqlTargetPlatform.AzureSqlManagedInstance.ToString());
+                string recommendationMIFileName = String.Format("SkuRecommendation{0}-{1}.json", AzureSqlTargetPlatform.AzureSqlManagedInstance.ToString(), timeStamp);
+                ExportRecommendationResultsAction.ExportRecommendationResults(reportMI, SqlAssessmentConfiguration.ReportsAndLogsRootFolderPath, false, recommendationMIFileName);
+                skuRecommendationFileNames[0] = recommendationMIFileName + SkuRecommendationJsonFileExt;
+                skuRecommendationFileNames[1] = recommendationMIFileName + SkuRecommendationHtmlFileExt;
+
+                SkuRecommendationReport reportVM = new SkuRecommendationReport(
+                    new Dictionary<SqlInstanceRequirements, List<SkuRecommendationResult>> { [parameters.InstanceRequirements] = parameters.SqlVmRecommendationResults },
+                    AzureSqlTargetPlatform.AzureSqlManagedInstance.ToString());
+                string recommendationVMFileName = String.Format("SkuRecommendation{0}-{1}.json", AzureSqlTargetPlatform.AzureSqlVirtualMachine.ToString(), timeStamp);
+                ExportRecommendationResultsAction.ExportRecommendationResults(reportVM, SqlAssessmentConfiguration.ReportsAndLogsRootFolderPath, false, recommendationVMFileName);
+                skuRecommendationFileNames[0] = recommendationVMFileName + SkuRecommendationJsonFileExt;
```

Comment:
`recommendationVMFileName` already is `SkuRecommendation{0}-{1}.json` -- wouldn't this be adding the file extension a second time?